docsearch/views/base.py

```diff
@@ -146,6 +146,26 @@ def form_valid(self, form):
 
     def get_queryset(self):
         sqs = super().get_queryset().models(self.model)
+
+        q = self.request.GET.get('q')
+        if q:
+            """
+            Boost results that contain direct matches of each term in the query,
+            regardless of whether any terms are enclosed in double quotes.
+            Terms in double quotes are exact match terms, so give them a bit of
+            an extra boost, and boost them as is - whitespace and all.
+            """
+            exact_match_re = re.compile(r'"(?P<phrase>.*?)"')
+            tokens = exact_match_re.split(q)
+            exacts = exact_match_re.findall(q)
+
+            for t in tokens:
+                if t and not t.strip().startswith("-"):
+                    if t in exacts:
+                        sqs = sqs.boost(t, 1)
+                    else:
+                        sqs = sqs.boost(t, .5)
```

Comment:
This is the only relevant change.

The docs for haystack say that setting a boost above 1.0 increases scores, while less than that decreases it. I have noticed that's not the case here, and any positive amount increases. It's probably because we're using an older version.

Also, this change has made it so that exact match queries (wrapped in double quotes) no longer omit anything that isn't an exact match. However, exact matches are still at the top of results. I'm a bit green to haystack and solr so if there's anything obvious we can do about that, then I'm down. Otherwise we might just be able to ask them if that's a big deal.